lib/lte_link_control/lte_lc.c

```diff
@@ -493,66 +496,61 @@ static void at_handler(void *context, const char *response)
 static int enable_notifications(void)
 {
 	int err;
-	char buf_sub[35];
 
 	/* +CEREG notifications, level 5 */
-	err = at_cmd_write(cereg_5_subscribe, NULL, 0, NULL);
+	err = nrf_modem_at_printf(AT_CEREG_5);
 	if (err) {
 		LOG_ERR("Failed to subscribe to CEREG notifications");
 		return err;
 	}
 
 	if (IS_ENABLED(CONFIG_LTE_LC_TAU_PRE_WARNING_NOTIFICATIONS)) {
-		snprintk(buf_sub,
-			 sizeof(buf_sub),
-			 AT_XT3412_SUB,
-			 CONFIG_LTE_LC_TAU_PRE_WARNING_TIME_MS,
-			 CONFIG_LTE_LC_TAU_PRE_WARNING_THRESHOLD_MS);
-
-		/* %XT3412 notifications subscribe */
-		err = at_cmd_write(buf_sub, NULL, 0, NULL);
+		err = nrf_modem_at_printf(AT_XT3412_SUB,
+					  CONFIG_LTE_LC_TAU_PRE_WARNING_TIME_MS,
+					  CONFIG_LTE_LC_TAU_PRE_WARNING_THRESHOLD_MS);
 		if (err) {
-			LOG_WRN("%s failed (%d), TAU pre-warning notifications are not enabled",
-				log_strdup(buf_sub), err);
-			LOG_WRN("%s is supported in nRF9160 modem >= v1.3.0", log_strdup(buf_sub));
+			LOG_WRN("Enabling TAU pre-warning notifications failed, error: %d", err);
+			LOG_WRN("TAU pre-warning notifications require nRF9160 modem >= v1.3.0");
 		}
 	}
 
 	if (IS_ENABLED(CONFIG_LTE_LC_MODEM_SLEEP_NOTIFICATIONS)) {
-		snprintk(buf_sub,
-			 sizeof(buf_sub),
-			 AT_XMODEMSLEEP_SUB,
-			 CONFIG_LTE_LC_MODEM_SLEEP_PRE_WARNING_TIME_MS,
-			 CONFIG_LTE_LC_MODEM_SLEEP_NOTIFICATIONS_THRESHOLD_MS);
-
 		/* %XMODEMSLEEP notifications subscribe */
-		err = at_cmd_write(buf_sub, NULL, 0, NULL);
+		err = nrf_modem_at_printf(AT_XMODEMSLEEP_SUB,
+					  CONFIG_LTE_LC_MODEM_SLEEP_PRE_WARNING_TIME_MS,
+					  CONFIG_LTE_LC_MODEM_SLEEP_NOTIFICATIONS_THRESHOLD_MS);
 		if (err) {
-			LOG_WRN("%s failed (%d), modem sleep notifications are not enabled",
-				log_strdup(buf_sub), err);
-			LOG_WRN("%s is supported in nRF9160 modem >= v1.3.0", log_strdup(buf_sub));
+			LOG_WRN("Enabling modem sleep notifications failed, error: %d", err);
+			LOG_WRN("Modem sleep notifications require nRF9160 modem >= v1.3.0");
 		}
 	}
 
 	/* +CSCON notifications */
-	err = at_cmd_write(cscon, NULL, 0, NULL);
+	err = nrf_modem_at_printf(cscon);
 	if (err) {
 		char buf[50];
 
+		/* Ensure null-termination of buffer */
+		buf[sizeof(buf) - 1] = '\0';
+
```

Comment:
I think the response from nrf_modem_at_cmd() must always be null terminated, otherwise the caller would not know the length of the response. So, there should be no need to add termination to the buffer?

Reply:
Yeah, I agree that it should be like that, but docs didn't say, so I treated it as if `snprintf` is used.
@lemrey Any input on this?

Reply:
All AT command responses coming from the modem are `NULL`-terminated and libmodem copies the whole response in to the buffer, including the `NULL` character.

Reply:
Good to know, thanks!